### ISSUE TITLE
[L10n] Leo UI is not mirrored for RTL

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#36308]
## Description 

Leo UI is not mirrored for RTL.

## Steps to reproduce 

   1. Change device language to any RTL, for example, Arabic or Hebrew
   2. Launch Brave
   3. Three-dot menu > Leo (if there is no Leo, enable it via the flag)
   4. *Observe that Leo UI is not mirrored for RTL*

## Actual result 

Leo UI is not mirrored for RTL. Leo's UI in the settings looks correct.

|Arabic|Hebrew|
|:-:|:-:|
| ![1](https://github.com/brave/brave-browser/assets/128532965/684f38d9-8b6c-4f1b-8ff8-8b51c42e66f1) | ![2](https://github.com/brave/brave-browser/assets/128532965/fd84e411-4113-43ee-b5ca-311f628b29ff) |
| ![1](https://github.com/brave/brave-browser/assets/128532965/7efe0465-ed4f-4fe6-9f83-c8cb18c570c0) | ![2](https://github.com/brave/brave-browser/assets/128532965/310b1b19-8e69-4971-86ba-67fa1e05d8fc) |
| ![1](https://github.com/brave/brave-browser/assets/128532965/92f3cb86-5930-4f55-ac64-3e3588f5daa2) | ![2](https://github.com/brave/brave-browser/assets/128532965/9d5cb9e4-e4a3-433d-8ec4-ac5aa6abcce9) |

## Expected result

Leo UI must be mirrored for RTL.

## Issue reproduces how often 

`Easily reproduced`

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? YES
- Can you reproduce this issue with the current Play Store Beta version? YES
- Can you reproduce this issue with the current Play Store Nightly version? YES

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): ANY
   - Android version: ANY
   
## Brave version

```
Brave build: 1.63.161
Chromium: 122.0.6261.57 (Official Build) (64-bit)
```

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by kjozwiak** [comment]
The above requires `1.63.165` or higher `1.63.x` verification 👍

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 14` using the following build(s):

```
Brave | 1.63.165 Chromium: 122.0.6261.94 (Official Build) (64-bit)
--- | ---
Revision | fed0fcc915f6d9e1db871202f0304c8d582a73de
OS | Android 14; Build/AP21.240119.009; 34; REL
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/36308#issue-2149834810, ensured that RTL was being respected as per the following:

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------
![1000001503](https://github.com/brave/brave-browser/assets/2602313/dea08949-c13a-4b94-9a0e-42dc4ea899bb) | ![1000001504](https://github.com/brave/brave-browser/assets/2602313/407476a5-c677-45e1-9395-71cb836b3a29) | ![1000001505](https://github.com/brave/brave-browser/assets/2602313/05500749-011c-4854-9240-141eebead82a) | ![1000001506](https://github.com/brave/brave-browser/assets/2602313/c83200ef-a851-4486-9484-8aa364bfd5b5)

Ran into https://github.com/brave/brave-browser/issues/36413 which @hffvld already filed and decided we'll fix the issue in the next minor `1.63.x` release as discussed via https://bravesoftware.slack.com/archives/C05NDQP03AQ/p1709079965402799.
